src/Microsoft.TestPlatform.Common/ExtensionFramework/DataCollectionExtensionManager.cs

```diff
@@ -75,18 +73,22 @@ public static DataCollectorExtensionManager Create(IMessageLogger messageLogger)
     /// <returns>
     /// The DataCollectorExtensionManager.
     /// </returns>
-    public static DataCollectorExtensionManager Create(string extensionAssemblyFilePath, bool skipCache, IMessageLogger messageLogger)
+    public static DataCollectorExtensionManager Create(string extensionAssemblyFilePath, bool skipCache, IMessageLogger messageLogger,
+        TestPluginCache? testPluginCache = null)
```

Comment:
Let the possibility to the caller to either use the default plugin cache (by passing null) or a custom one. Note that a plugin cache wraps a "load context".